classes/02class/exercises/c02-network04/alanlima/README.md

```diff
@@ -0,0 +1,204 @@
+    --resources
+
+# Creating Subnet: private-c ap-southeast-2c 172.16.22.0/24 vpc-0ab8012d8ba301c77
+$ aws ec2 create-subnet \
+    --availability-zone ap-southeast-2c \
+    --cidr-block 172.16.22.0/24 \
+    --vpc-id vpc-0ab8012d8ba301c77 \
+    --output text \
+    --query 'Subnet.SubnetId' \
+    | xargs aws ec2 create-tags \
+    --tags Key=Name,Value=private-c \
+    --resources
+
+# Create Internet Gateway for the public subnets
+
+$ aws ec2 create-internet-gateway
+InternetGateway:
+  Attachments: []
+  InternetGatewayId: igw-084a30d9182d99964
+  OwnerId: '097922957316'
+  Tags: []
+
+# Attach the IGW to the VPC
+$ aws ec2 attach-internet-gateway \
+    --vpc-id vpc-0ab8012d8ba301c77 \
+    --internet-gateway-id igw-084a30d9182d99964
+
+# Create the route table which will allow connection to IGW
+$ aws ec2 create-route-table \
+    --vpc-id vpc-0ab8012d8ba301c77
+    
+RouteTable:
+  Associations: []
+  OwnerId: '097922957316'
+  PropagatingVgws: []
+  RouteTableId: rtb-097219c81a1e1c641
+  Routes:
+  - DestinationCidrBlock: 172.16.0.0/16
+    GatewayId: local
+    Origin: CreateRouteTable
+    State: active
+  Tags: []
+
+# Map any external connection to the gateway
+$ aws ec2 create-route \
+        --route-table-id rtb-097219c81a1e1c641 \
+        --destination-cidr-block 0.0.0.0/0 \
+        --gateway-id igw-084a30d9182d99964
+
+# Check the routes
+$ aws ec2 describe-route-tables --route-table-id rtb-097219c81a1e1c641 
+RouteTables:
+- Associations: []
+  OwnerId: '097922957316'
+  PropagatingVgws: []
+  RouteTableId: rtb-097219c81a1e1c641
+  Routes:
+  - DestinationCidrBlock: 172.16.0.0/16
+    GatewayId: local
+    Origin: CreateRouteTable
+    State: active
+  - DestinationCidrBlock: 0.0.0.0/0
+    GatewayId: igw-084a30d9182d99964
+    Origin: CreateRoute
+    State: active
+  Tags: []
+  VpcId: vpc-0ab8012d8ba301c77
+
+# Associate the created routet table to the public subnets
+
+$ aws ec2 describe-subnets \
+    --filters "Name=tag:Name,Values=public-a,public-b,public-c" \
+    --output text \
+    --query 'Subnets[*].{Id:SubnetId}' \
+    | xargs -I % sh -c "echo Associating route to %; aws ec2 associate-route-table --subnet-id % --route-table-id rtb-097219c81a1e1c641"
+
+Associating route to subnet-0d0ccd942d0d65099
+AssociationId: rtbassoc-00f635cb720a51bc9
+AssociationState:
+  State: associated
+Associating route to subnet-05c30e57750cccbc9
+AssociationId: rtbassoc-0f916f423a0a3e22a
+AssociationState:
+  State: associated
+Associating route to subnet-07a8fa273813e36f6
+AssociationId: rtbassoc-009db4028ee49bcc7
+AssociationState:
+  State: associated
+
+# Modify the subnets to map public ip for new nodes inside the public subnets
+$ aws ec2 describe-subnets \
+    --filters "Name=tag:Name,Values=public-a,public-b,public-c" \
+    --output text \
+    --query 'Subnets[*].{Id:SubnetId}' \
+    | xargs -I % sh -c "echo Modifying subnet to %; aws ec2 modify-subnet-attribute --subnet-id % --map-public-ip-on-launch"
```

Comment:
this seems to be part of future labs...

Reply:
Yeah, once that I started to read the following exercises I see that I already did.
I will send a new commit removing those extra scripts.